aas-web-ui/src/components/SubmodelPlugins/PCF.vue

```diff
@@ -0,0 +1,173 @@
+        },
+        colorPercentage() {
+            return this.calculatePercentage(this.PCFCO2eqTotal, this.TCFCO2eqTotal);
+        },
+        pcfValueColor() {
+            return this.getThemeColor('pcf');
+        },
+        tcfValueColor() {
+            return this.getThemeColor('tcf');
+        },
+        fingersColor() {
+            return this.getThemeColor('fingers');
+        },
+        PCFCO2eqTotal(): number {
+            return this.PCFCO2eq.reduce((acc, item) => acc + item.value, 0);
+        },
+        TCFCO2eqTotal(): number {
+            return this.TCFCO2eq.reduce((acc, item) => acc + item.value, 0);
+        }
+    },
+    methods: {
+        calculatePercentage(productValue: number, transportValue: number): number {
+            const total = productValue + transportValue;
+            return total ? (productValue / total) * 100 : 0;
+        },
+        getThemeColor(colorKey: string): string {
+            const isDark = this.$vuetify.theme.global.current.dark;
+            const themeColors = isDark ? this.$vuetify.theme.themes.dark.colors : this.$vuetify.theme.themes.light.colors;
+            return themeColors[colorKey];
+        },
+        initializeCarbonFootprint(semanticId: string, value: 'PCFCO2eq' | 'TCFCO2eq') {
+            // console.log('Component Mounted. SubmodelElementData:', this.submodelElementData);
+            if (Object.keys(this.submodelElementData).length === 0) {
+                return;
+            }
+            try {
+                const carbonFootprintData = { ...this.submodelElementData };
+                const entries = carbonFootprintData.submodelElements.filter((elem: any) => {
+                    return elem.semanticId?.type === 'ExternalReference' &&
+                        elem.semanticId?.keys?.[0]?.type === 'GlobalReference' &&
+                        elem.semanticId?.keys?.[0]?.value === semanticId;
+                });
+                if (entries.length === 0) {
+                    console.warn(`No entries found for ${semanticId}.`);
+                    return;
+                }
+                this[value] = entries.map((entry: any) => ({
+                    semanticId: semanticId,
+                    value: parseFloat(entry.value?.find((val: any) => val?.idShort?.includes('CO2eq'))?.value) || 0
+                }));
+            } catch (error) {
+                console.error(`Error initializing ${semanticId}:`, error);
+            }
+        },
+        initializeProductCarbonFootprint() {
+            this.initializeCarbonFootprint('https://adminshell.io/idta/CarbonFootprint/ProductCarbonFootprint/0/9', 'PCFCO2eq');
+        },
+        initializeTransportCarbonFootprint() {
+            this.initializeCarbonFootprint('[IRDI] https://adminshell.io/idta/CarbonFootprint/TransportCarbonFootprint/0/9', 'TCFCO2eq');
+        },
+        unitSuffix(value: Array<any>): string {
+            return value.length > 0 && value[0].unit || '';
+        }
```

Comment:
In your example AASX files there is missing the Unit for the PCFCO2eq resp. TCFCO2eq properties. The unit can be specified via embeddedDatapSecification, e.g.:
![PCFCO2eq_Example](https://github.com/user-attachments/assets/c4a6d4b9-fa8e-4328-b728-4da2caeeaef0)

Then you can get the unit of the property via `unitSuffix(prop: any)` (see [SubmodelElementHandling.ts](https://github.com/eclipse-basyx/basyx-aas-web-ui/blob/0f2532f17f36c4c8d74d93c69b79d595cdc38faa/aas-web-ui/src/mixins/SubmodelElementHandling.ts#L579))